app/models/thredded/user_permissions/write/all.rb

```diff
@@ -9,7 +9,8 @@ module All
 
         # @return [ActiveRecord::Relation<Thredded::Messageboard>] messageboards that the user can post in
         def thredded_can_write_messageboards
-          Thredded::Messageboard.all
+          # Thredded::Messageboard.all
+          Thredded::Messageboard.where(locked: false)
```

Comment:
I believe it's better to revert the changes in this file and do the check on the `MessageboardPolicy` level.

This is because we must filter `PostPolicy#create?` based on this method, but we want to allow posting to the existing topics.

This filtering already happens via `MessageboardPolicy#post?` but for this we'll replace that with `@user.thredded_can_write_messageboards.include?(@messageboard)` in `PostPolicy`.